### DISCUSSION TITLE
Renaming columns

### DISCUSSION THREAD

**Discussion by nickathot** [TabularEditor/TabularEditor3#1353] [🙋♀️TE3 Q&A] [answered]
Hi
I created a calculated table with 2 columns. The columns were created with default names of 'Value1' and 'Value2'.
How can you rename these columns as the name property is greyed out?

Or how can you create calculated table with specific column names?
![2024-07-24 16_03_10-Window](https://github.com/user-attachments/assets/e336e2e1-008d-42c0-b351-c5c4c4916743)

**Comment by mlonsk** [comment]
Hi @nickathot 
You need to change the DAX Expression where the columns are named. 

![image](https://github.com/user-attachments/assets/f5e7294b-b5e1-4ae0-b9aa-50d9fd5afcaa)

The names of the columns are defined in the DAX Expression, which is why the Name property is greyed out.

**Reply by nickathot** [reply]
Hi @mlonsk 

Thank you for your reply.

Unfortunately, that solution did not work as I require multiple rows.
Luckily for my colleague he came up with a alternative solution using DATATABLE

**Comment by nickathot** [comment, marked as answer]
To create a calculated table with multiple columns and rows with custom column names use DATATABLE, refer to image.

![2024-07-25 09_50_51-Window](https://github.com/user-attachments/assets/88137c1f-b6d9-4d9f-9799-ca9827c93aca)

**Reply by mlonsk** [reply]
Neat solution. Mine was merely an example of how you can set the name through the DAX statement, but as @otykier said in [his response](https://github.com/TabularEditor/TabularEditor3/discussions/1353#discussioncomment-10145241), there is another way where you set the property "Name Inferred" on the column to false and can then change the name property:

![image](https://github.com/user-attachments/assets/3abecc23-16fc-4190-8a8f-4ef0101d7bc1)

**Comment by otykier** [comment]
You can rename columns of a calculated table without modifying the DAX, by setting the “Is Name Inferred” property of the column to “false”. Similarly, if you need to change the data type of a column (without modifying the DAX), you can set “Is Data Type Inferred” to “false” (this goes for calculated columns as well).
